### ISSUE TITLE
When tC is upgraded, do not delete any Bible resources that will not be replaced during the upgrade

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/translationCore#6145]
- [x] When updating tA, tN, or tW, or any Greek/Hebrew helps, the older version must be deleted
- [ ] (M) When installing a new version of tC, any imported Bible resources (e.g. `t4t`, or Russian `ROB`,`RSB`, etc. ) need to be retained
  - [ ] When an imported Bible is updated, delete the old version (@PhotoNomad0 - except for ugnt and hbo where we may need to retain old versions for tN)
  - [ ] 

===

Reported by Robert on 1.1.4
As a checker or aligner I do not want to lose resources when upgrading tC so that I can easily continue my work without having to figure out why I've lost and how to recover resources I was using.

Robert uses the T4T English resource in his work. After upgrading from 1.1.3 to 1.1.4 he had to update his English resources in order to get it in his app again.

**Comment by PhotoNomad0** [comment]
This is a very dangerous practice, since recently we have been making changes to how we store the content in the resources folder.  Additional TN and TW content that was downloaded needs to be downloaded again and run through the  updated source content parser.  This means that keeping old content could cause breakage.  It may work if we only preserve Bible resources. @benjore @cckozie

**Comment by benjore** [comment]
Need to figure out how to prevent old/antiquated file formats from breaking tC.

**Comment by PhotoNomad0** [comment]
> When updating tA, tN, or tW, the older version must be deleted

@benjore - I think all of the tHelps that are not updated should be deleted.  Since we don’t bundle all the helps in the catalog in tCore, I think leaving around any tA, tN, tW, etc. that was imported with an older version of tCore may be a ticking time bomb - either we will have to write migrations when we change storage formats or need to direct the user to re-download them so to be safe.  Maybe @mannycolon is right that we also need to save the version of the source content updater that processed it.  But that could start getting complicated quickly.

**Comment by PhotoNomad0** [comment]
Verified in translationCore 1.2.0 (ca7a093)

**Comment by cckozie** [comment]
1.2.0 (89d7abe)
Most of the existing resources are persisted during an upgrade to 1.2, but there are three issues:
~~ISSUE 1:~~ #6235 was created for this problem
~~The translationHelps folders are lost for all non-packaged languages~~
ISSUE 2:
After the upgrade there are two different Greek resource folders; grc (left over from 1.1.4), and el-x-koine added in 1.2. This causes the UGNT to appear twice in the resource dropdown for the scripture pane.
We may want to provide a migration to move any additional Greek texts to the el-x-koine folder and remove the grc folder. @benjore ??

![image](https://user-images.githubusercontent.com/8171759/61063023-e3aa0a00-a3c4-11e9-8057-f35b5b66a709.png)

ISSUE 3:
Running 1.2, I had UGNT v0.7 and v0.8 both on the local file system. I deleted v0.8 and restarted tC. Now I have only v0.8 on the system, v0.7 is gone. 
Verified in a retest when there is a project that requires a previous version of the UGNT, after deleting the v0.8 folder and restarting tC, the older version of the UGNT does get deleted.

resources in 1.1.4: ------------------------------------   resources after upgrading to 1.2:
<img width="328" alt="resources12" src="https://user-images.githubusercontent.com/8171759/61063221-31267700-a3c5-11e9-8035-ad78591d1ddf.png"><img width="321" alt="rersources114" src="https://user-images.githubusercontent.com/8171759/61063179-1ce27a00-a3c5-11e9-9366-eb9c1c1982ed.png">

**Comment by PhotoNomad0** [comment]
@benjore @cckozie - how should we word things on this new popup.  something like "Helps resources on your system are not compatible with this new version of translationCore and need to be downloaded again: ru, es-419, ...
Do you want to download them now?"

That's the best I could think of now.

**Comment by PhotoNomad0** [comment]
Nevermind - I guess that would be on new issue.

**Comment by PhotoNomad0** [comment]
Verified in translationCore 2.0.0 (1d9a4c5)

**Comment by cckozie** [comment]
Still seeing ISSUE 2 and ISSUE 3 failing in 2.0.0 (00bca0b).
I think the problem I am seeing with ISSUE 3 is because I had done an intermediate upgrade to a 1.2 build in order to get an older UGNT downloaded to support a tN project. That is probably not a legitimate test since I don't think there can be any instance where prior to 2.0 a user will have multiple versions of  the UGNT on their system.
I retested with multiple Greek texts and some non-packaged languages, and it does work as expected other than ISSUE 1 which has been moved to a separate issue.
